openapi.yml

```diff
@@ -32,65 +51,74 @@ paths:
                 type: array
                 items:
                   $ref: "#/components/schemas/OrderFull"
-  /api/v1/nonce:
+  /api/v1/gas-tip/{sellToken}:
     get:
-      summary: The nonce to pick for a new order.
-      description: The best nonce to pick depends on how much time there is left in the current settlement according to the operator. This time is token-pair specific which is why the sell and buy tokens must be supplied.
-      parameters:
-        - name: sellToken
-          in: query
-          schema:
-            $ref: "#/components/schemas/Address"
-        - name: buyToken
-          in: query
-          schema:
-            $ref: "#/components/schemas/Address"
+      summary: The tip that is used to pay solvers for their service in the sellToken of an order.
+      description: The tip is calculated by the order service and depends on the current gas price and the expected gas costs to settle the trade
       responses:
         200:
-          description: the nonce
+          description: the tip
           content:
             application/json:
               schema:
-                $ref: "#/components/schemas/Nonce"
+                $ref: "#/components/schemas/Tip"
+        400:
+          description: sellToken not existent
 components:
   schemas:
     Address:
       description: Ethereum 40 byte address encoded as a hex without `0x` prefix.
       type: string
       example: "6810e776880c02933d47db1b9fc05908e5386b96"
     TokenAmount:
-      description: Amount of a token. uint112 encoded in decimal.
+      description: Amount of a token. uint256 encoded in decimal.
       type: string
       example: "1234567890"
-    Nonce:
-      description: Nonce of the batch in which the order is valid. uint32.
-      type: integer
+    Tip:
+      description: The tip that is used to pay solvers. uint256.
+      type: object
+      properties:
+        expirationDate:
+          description: Expiration date of the offered tip. Encoded as ISO 8601 UTC.
+          type: string
+          example: "2020-10-22T07:57:18Z"
+        tipAmount:
+          description: Amount of tip in specified sellToken
+          type: integer
+    OrderType:
+      description: Is this a buy order or sell order?
+      type: string
+      enum: [buy, sell]
     OrderNew:
       description: Data a user provides when creating a new order.
       type: object
       properties:
```

Comment:
ubernit: We should probably order the order properties to match the order in which they are hashed, that is:
```
export const ORDER_TYPE_FIELDS = [
  { name: "sellToken", type: "address" },
  { name: "buyToken", type: "address" },
  { name: "sellAmount", type: "uint256" },
  { name: "buyAmount", type: "uint256" },
  { name: "validTo", type: "uint32" },
  { name: "appData", type: "uint32" },
  { name: "tip", type: "uint256" },
  { name: "kind", type: "uint8" },
  { name: "partiallyFillable", type: "bool" },
];
```